### ISSUE TITLE
`unicorn/better-regex` autofix breaks "match all" in multiline regexps 

### ISSUE THREAD

**Issue by FloEdelmann** [sindresorhus/eslint-plugin-unicorn#895]
This regexp is autofixed by `unicorn/better-regex` like this:

```diff
- /^lorem(?:.|\n)*?ipsum/m
+ /^lorem[.\n]*?ipsum/m
```

This breaks the regex:

* `(.|\n)` means "any character, including a newline character"
* `[.\n]` means "a period character or a newline character"

Instead, it should be fixed to `[^]`, according to https://stackoverflow.com/a/16119722/451391.

**Comment by RunDevelopment** [comment]
While this is definitely a bug, your proposed expected fix is incorrect. In JS RegExp, the dot (`.`) is equivalent to `[^\r\n\u2028\u2029]` (unless the `s` flag is enabled), so `(?:.|\n)` is equivalent to `[^\r\u2028\u2029]`. So the correct fix would be:

```diff
- /lorem(?:.|\n)*?ipsum/m
+ /lorem[^\r\u2028\u2029]*?ipsum/m
```

(Whether this fix is desirable is a different question.)

**Comment by sindresorhus** [comment]
This should be reported at https://github.com/DmitrySoshnikov/regexp-tree, which is what does the regex transformations.

**Comment by FloEdelmann** [comment]
I've opened an upstream issue: https://github.com/DmitrySoshnikov/regexp-tree/issues/217

**Comment by sindresorhus** [comment]
It was removed: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/89dc5dc04389929ae786e171bac2af522fe2e7d8
